File: archive/f/f-sharp/EvenOdd.fs
open System

let usage = "Usage: please input a number"

module EvenOdd =
    let run n = if n % 2 = 0 then "Even" else "Odd"

module Helpers =
    let (|Int|_|) (s: string) =
        match s.Trim() |> Int32.TryParse with
        | true, n -> Some n
        | false, _ -> None

    let parseArgs argv =
        match argv with
        | [| Int n |] -> Ok n 
        | _ -> Error usage

    let handleResults =
        function
        | Ok result ->
            printfn "%s" result
            0
        | Error msg ->
            eprintfn "%s" msg
            1

[<EntryPoint>]
let main argv =
    argv
    |> Helpers.parseArgs
    |> Result.map EvenOdd.run
    |> Helpers.handleResults

File: archive/v/visual-basic/even-odd.vb
Module EvenOdd

    Public Sub Main(args As String())

        If args.Length = 0 Then
            Console.WriteLine("Usage: please input a number")
            Environment.Exit(1)
        End If

        Dim n As Integer

        If Not Integer.TryParse(args(0), n) Then
            Console.WriteLine("Usage: please input a number")
            Environment.Exit(1)
        End If

        Dim result As String = If(n Mod 2 = 0, "Even", "Odd")

        Console.WriteLine(result)

    End Sub

End Module